torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -1454,11 +1468,20 @@ void LLVMCodeGenImpl::visit(const FunctionCall* v) {
 }
 
 void LLVMCodeGenImpl::visit(const Allocate* v) {
-  throw unimplemented_lowering(v);
+  llvm::Value* size =
+      llvm::ConstantInt::getSigned(LongTy_, v->dtype().byte_size());
+  for (const Expr* e : v->dims()) {
+    e->accept(this);
+    size = irb_.CreateMul(size, irb_.CreateZExt(value_, LongTy_));
+  }
+
+  value_ = irb_.CreateAlloca(dtypeToLLVM(v->dtype()), size);
```

Comment:
perhaps best to hardcode a threshold on using alloca (including a requirement that size can be simplified to a constant)